### ISSUE TITLE
InferenceService Deployment with ModelMesh Example: fail to send requests from Istio Ingress

### ISSUE THREAD

**Issue by IntelliJJ824** [kserve/kserve#2212]
/kind bug

**What steps did you take and what happened:**
I follow the InferenceService Deployment with ModelMesh: example-sklearn-isvc from [here](https://github.com/kserve/modelmesh-serving/blob/main/docs/inferenceservice.md)

Here is what I have already get, and it has been tested successfully by using the port-forward method as the example instruction.

_**However, I am considering to achieve serverless function by curl from Ingress gateway with HOST Header.
I have tried different curl commands, but none of them work.**_
`curl -v -H "Host: ${SERVICE_HOSTNAME}" http://${INGRESS_HOST}:${INGRESS_PORT}/v1/models/example-sklearn-isvc:predict -d @./isvc-input.json`

`curl -v -H "Host: ${SERVICE_HOSTNAME}" http://${INGRESS_HOST}:${INGRESS_PORT}/v1/models/example-sklearn-isvc/infer -d @./isvc-input.json `

They all show the 404 Not found in the following:
`Trying 192.168.49.2:30413...
 TCP_NODELAY set
 Connected to 192.168.49.2 (192.168.49.2) port 30413 (#0)
 POST /v1/models/example-sklearn-isvc/infer HTTP/1.1
Host: modelmesh-serving.modelmesh-serving
User-Agent: curl/7.68.0
Content-Length: 431
Content-Type: application/x-www-form-urlencoded
 
 upload completely sent off: 431 out of 431 bytes
 Mark bundle as not supporting multiuse
 HTTP/1.1 404 Not Found
 date: Thu, 02 Jun 2022 20:53:01 GMT
 server: istio-envoy
 connection: close
 content-length: 0
 Closing connection 0`

**Environment:**

- Istio Version: v1.13.4
- Knative Version:v1.4.0
- KFServing Version:v0.7.0
- Kubeflow version: no use
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube version: v1.25.2
- Kubernetes version: (use `kubectl version`): v1.24.1

Could you figure out where the problem is? Thanks

**Comment by IntelliJJ824** [comment]
Just a following question: Does KServe v0.8.0 support ModelMesh at this moment? I have seen the example-sklearn-isvc still in KServe v0.7.0 schema.

**Comment by yuzisun** [comment]
yes it is supported, @pvaneck I think he needs to enable the rest proxy to make the curl to work right?

**Comment by IntelliJJ824** [comment]
I can see the URL is still in GRPC `grpc://modelmesh-serving.modelmesh-serving:8033`. Any hints on how to set the rest proxy for it?

**Comment by IntelliJJ824** [comment]
Hi, any suggestion on Modelmesh with using Istio + KNative as the entry point for calling up the replicas?
